adapters/impactify/impactify.go

```diff
@@ -0,0 +1,198 @@
+		err := json.Unmarshal(bidRequest.Imp[i].Ext, &defaultExt)
+		if err != nil {
+			return nil, []error{&errortypes.BadServerResponse{
+				Message: "Unable to decode the imp ext",
+			}}
+		}
+
+		impactifyExt.Impactify = defaultExt.Bidder
+		bidRequest.Imp[i].Ext, err = json.Marshal(impactifyExt)
+		if err != nil {
+			return nil, []error{&errortypes.BadServerResponse{
+				Message: "Unable to decode the imp ext",
+			}}
+		}
+
+		validImps = append(validImps, bidRequest.Imp[i])
+	}
+
+	if len(validImps) == 0 {
+		if errs != nil {
+			return nil, errs
+		} else {
+			return nil, []error{&errortypes.BadInput{
+				Message: "No valid impressions in the bid request",
+			}}
+		}
+	}
+
+	bidRequest.Imp = validImps
+
+	reqJSON, err := json.Marshal(bidRequest)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+	if bidRequest.Device != nil {
+		if bidRequest.Device.UA != "" {
+			headers.Add("User-Agent", bidRequest.Device.UA)
+		}
+		// Add IPv4 or IPv6 if available
+		if bidRequest.Device.IP != "" {
+			headers.Add("X-Forwarded-For", bidRequest.Device.IP)
+		} else if bidRequest.Device.IPv6 != "" {
+			headers.Add("X-Forwarded-For", bidRequest.Device.IPv6)
+		}
+	}
+	if bidRequest.Site != nil {
+		headers.Add("Referer", bidRequest.Site.Page)
+	}
+
+	// set user's cookie
+	if bidRequest.User != nil && bidRequest.User.BuyerUID != "" {
+		headers.Add("Cookie", "uids="+bidRequest.User.BuyerUID)
+	}
+
+	adapterRequests = append(adapterRequests, &adapters.RequestData{
+		Method:  "POST",
+		Uri:     a.endpoint,
+		Body:    reqJSON,
+		Headers: headers,
+	})
+
+	return adapterRequests, errs
+}
+
+func (a *adapter) MakeBids(
+	internalRequest *openrtb2.BidRequest,
+	externalRequest *adapters.RequestData,
+	response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	} else if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: "Invalid request.",
+		}}
+	} else if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected HTTP status %d.", response.StatusCode),
+		}}
+	}
+
+	var openRtbBidResponse openrtb2.BidResponse
+
+	if err := json.Unmarshal(response.Body, &openRtbBidResponse); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: "Bad server body response",
+		}}
+	}
+
+	if len(openRtbBidResponse.SeatBid) == 0 {
+		return nil, nil
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(openRtbBidResponse.SeatBid[0].Bid))
+	bidResponse.Currency = openRtbBidResponse.Cur
+
+	sb := openRtbBidResponse.SeatBid[0]
+	for i := 0; i < len(sb.Bid); i++ {
+		if !(sb.Bid[i].Price > 0) {
+			continue
```

Comment:
Please add a test for this negative price condition. All tests include a valid price only.

Reply:
This test has been added.